optimum/intel/pot/quantize.py

```diff
@@ -0,0 +1,117 @@
+import os
+import glob
+import yaml
+import logging
+
+from .quantize_utils import OVQADataLoader, OVQAAccuracyMetric
+
+from compression.engines.ie_engine import IEEngine
+from compression.graph import load_model, save_model
+from compression.graph.model_utils import compress_model_weights
+from compression.pipeline.initializer import create_pipeline
+
+from addict import Dict
+
+LOGGER = logging.getLogger(__name__)
+
+OV_WEIGHTS_NAME = "ov_model.xml"
+
+class bcolors:
+    """
+    Just gives us some colors for the text
+    """
+    HEADER = '\033[95m'
+    OKBLUE = '\033[94m'
+    OKGREEN = '\033[92m'
+    WARNING = '\033[93m'
+    FAIL = '\033[91m'
+    ENDC = '\033[0m'
+    BOLD = '\033[1m'
+    UNDERLINE = '\033[4m'
+
+def get_data_loader_cls(config):
+    if config.task_type == "QuestionAnswering":
+        return OVQADataLoader(config)
+
+def get_metric_cls(config):
+    if config.task_type == "QuestionAnswering":
+        return OVQAAccuracyMetric(config)
+
+class OVAutoQuantizer():
+    def __init__(self, model_name: str, pretrained_model, config_path):
+        self.config = Dict(self.parse_config(config_path))
+        self.config.model_name = model_name 

```

Comment:
Can we extract `model_name` from the `pretrained_model`?

Reply:
Done